src/array.rs

```diff
@@ -0,0 +1,456 @@
+use std::ffi::c_void;
+
+use half::{bf16, f16};
+use mlx_sys::mlx_array;
+use num_complex::Complex;
+
+use crate::sealed::Sealed;
+
+// TODO: camel case?
+// Not using Complex64 because `num_complex::Complex64` is actually Complex<f64>
+#[allow(non_camel_case_types)]
+pub type c64 = Complex<f32>;
```

Comment:
My feeling is that `complex64` is probably better, just because unlike `i32, u8` or whatever which are ingrained in most devs head, c64 wouldn't be immediately understood.

But with IDEs it doesn't matter much I suppose. c64 is closer to `i32 and u8` let's leave it as such for now.

Reply:
Changed to `complex64`